### PR TITLE
Add preservation package specification

### PR DESCRIPTION
**Pull request by hakbailey**
#### Why these changes are being introduced:
To document the structure, contents, and metadata requirements for submitting published theses to preservation.

#### How this addresses that need:
* Adds preservation package specification file to the docs folder
* Adds an example bag following the specification

#### Relevant ticket(s):
* https://mitlibraries.atlassian.net/browse/ETD-503

#### Developer

- [ ] All new ENV is documented in README
- [ ] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [ ] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [ ] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

NO

### CODE REVIEW

File: docs/preservation-package-spec.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+See file [1721.1_123456-thesis.zip/data/metadata/metadata.csv](1721.1_123456-thesis.zip/data/metadata/metadata.csv) in this project's docs folder for an metadata file structured according to these requirements.
+
+All metadata about the thesis must be in a single CSV file with a column for each metadata field and a row for each file associated with the thesis.
+
+Thesis-level metadata  (title, abstract, author, etc.) should be in the row for the thesis PDF file. Repeated fields must have a column for each instance of the field in the metadata. Note: If a rights field includes the copyright character, it must be UTF-8 encoded in the CSV file for Archivematica to parse it correctly.
+
+Other files included in the CSV should have blank values for any fields not directly related to the file -- fields associated with the thesis such as title, abstract, author should be blank, but fields specific to the file such as BitstreamChecksumValue should have the value present in the row for that file.
+
+In addition to the metadata submitted to DSpace, some additional fields are required for presentation and should each have their own column in the CSV file. These are:
```
Comment (JPrevost) [inline docs/preservation-package-spec.md:31]: Is `presentation` supposed to be `preservation`?

File: docs/preservation-package-spec.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+## Metadata File
+
+See file [1721.1_123456-thesis.zip/data/metadata/metadata.csv](1721.1_123456-thesis.zip/data/metadata/metadata.csv) in this project's docs folder for an metadata file structured according to these requirements.
+
+All metadata about the thesis must be in a single CSV file with a column for each metadata field and a row for each file associated with the thesis.
+
+Thesis-level metadata  (title, abstract, author, etc.) should be in the row for the thesis PDF file. Repeated fields must have a column for each instance of the field in the metadata. Note: If a rights field includes the copyright character, it must be UTF-8 encoded in the CSV file for Archivematica to parse it correctly.
+
+Other files included in the CSV should have blank values for any fields not directly related to the file -- fields associated with the thesis such as title, abstract, author should be blank, but fields specific to the file such as BitstreamChecksumValue should have the value present in the row for that file.
```
Comment (JPrevost) [inline docs/preservation-package-spec.md:29]: In the example, there is no checksum provided for the signature page. Is the intent to only provide checksums for files we have submitted to DSpace? It seems useful to always include a checksum so archivematica can (in theory) confirm the file is exactly what we intended to submit even if we haven't sent it to DSpace.

File: docs/preservation-package-spec.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    - _last-kerberos-DEG-DEP-YYYY_-thesis.pdf The data folder must at minimum contain the PDF representation of the thesis
+    - _last-kerberos-DEG-DEP-YYYY-supplemental1.txt_ The data folder may additionally contain any other files submitted with the thesis including supplementary files, signature pages, etc
+  - bag-info.txt Per the bagit specfication, the bag must contain a bag-info.txt file with the required contents
+  - bagit.txt Per the bagit specfication, the bag must contain a bagit.txt file with the required contents
+  - manifest-_algorithm_.txt Per the bagit specfication, the bag must contain at least one manifest-_algorithm_.txt file containing the checksum and file path of each file in the data folder, where _algorithm_ in the file name should be the cryptographic hash algorithm use to create the checksums, e.g. "sha256"
+  - tagmanifest-_algorithm_.txt Per the bagit specfication, the bag must contain at least one tagmanifest-_algorithm_.txt file containing the checksum and file path of each file in the top-level bag folder, where _algorithm_ in the file name should be the cryptographic hash algorithm use to create the checksums, e.g. "sha256"
```
Comment (JPrevost) [inline docs/preservation-package-spec.md:19]: What? After reading this multiple times, I think it means we tagmanfest will contain hashes for `bag-info.txt`, `bagit.txt`, and `manifest-_algorithm_.txt`? Part of the confusion is that `tagmanifest-_algorithm_.txt` is also in the top level and can't contain it's own hash as adding that hash would change the hash...right?

File: docs/preservation-package-spec.md (hunk 2 above)
Comment (hakbailey) [reply docs/preservation-package-spec.md:29]: I forgot we had checksums for files we don't send to DSpace :). Based on our discussion yesterday, I agree we likely want to include the checksum-as-received for all files.

File: docs/preservation-package-spec.md (hunk 3 above)
Comment (hakbailey) [reply docs/preservation-package-spec.md:19]: Yeah I'm not sure how specific we need to be about those files as they just follow exactly the bagit spec. I don't think we need to reiterate that here? (And using bagit libraries, which we will do, creates those files for us.)

File: docs/preservation-package-spec.md (hunk 1 above)
Comment (hakbailey) [reply docs/preservation-package-spec.md:31]: Yes, definitely.

File: docs/preservation-package-spec.md (hunk 3 above)
Comment (JPrevost) [reply docs/preservation-package-spec.md:19]: Hmmm. I'm not confident there are well maintained bagit libraries for ruby. However, this is a separate conversation. I'm comfortable with leaving the details here or just doing a more general "follow the bagit spec".

**Review by JPrevost — APPROVED** [review]

**Review by JPrevost — APPROVED** [review]

File: docs/preservation-package-spec.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+- _dspace_handle_-thesis.zip The top-level folder must be a zipped bag named with the
+  thesis's DSpace handle followed by "-thesis.zip"
+  - data/ There must be a folder in the bag named "data", containing all files associated with the thesis
+    - metadata/ There must be a folder in the "data" folder named "metadata", containing exactly one file of the metadata for the thesis
+      - metadata.csv The metadata file must be named "metadata.csv". See [Metadata File section](#metadata-file) below for details on the contents of this file
+    - _last-kerberos-DEG-DEP-YYYY_-thesis.pdf The data folder must at minimum contain the PDF representation of the thesis
+    - _last-kerberos-DEG-DEP-YYYY-supplemental1.txt_ The data folder may additionally contain any other files submitted with the thesis including supplementary files, signature pages, etc
+  - bag-info.txt Per the bagit specfication, the bag must contain a bag-info.txt file with the required contents
```
[the remark this answers is not in the document]
Comment (hakbailey) [reply docs/preservation-package-spec.md]: I'm confused by this comment, that was updated in the last commit?

**Pull request merged**